media-video/ccextractor/patches/ccextractor-0.85.patchset

```diff
@@ -0,0 +1,87 @@
+From e3c0248b02899a0be0fa95a9d395c99526e00fa8 Mon Sep 17 00:00:00 2001
+From: sfanxiang <sfanxiang@gmail.com>
+Date: Mon, 8 Jan 2018 18:08:11 +0000
+Subject: port to Haiku
+
+
+diff --git a/linux/build b/linux/build
+index 6897901..f3e740f 100644
+--- a/linux/build
++++ b/linux/build
+@@ -1,6 +1,7 @@
+ #!/bin/bash
++INCLUDE_DIR=$(finddir B_SYSTEM_HEADERS_DIRECTORY)
+ BLD_FLAGS="-g -std=gnu99 -Wno-write-strings -DGPAC_CONFIG_LINUX -D_FILE_OFFSET_BITS=64 -DVERSION_FILE_PRESENT -DENABLE_OCR"
+-BLD_INCLUDE="-I../src -I /usr/include/leptonica/ -I /usr/include/tesseract/ -I../src/lib_ccx/ -I../src/gpacmp4/ -I../src/libpng/ -I../src/zlib/ -I../src/zvbi -I../src/lib_hash -I../src/protobuf-c -I../src/utf8proc"
++BLD_INCLUDE="-I../src -I $INCLUDE_DIR/leptonica/ -I $INCLUDE_DIR/tesseract/ -I../src/lib_ccx/ -I../src/gpacmp4/ -I../src/libpng/ -I../src/zlib/ -I../src/zvbi -I../src/lib_hash -I../src/protobuf-c -I../src/utf8proc"
+ SRC_LIBPNG="$(find ../src/libpng/ -name '*.c')"
+ SRC_ZLIB="$(find ../src/zlib/ -name '*.c')"
+ SRC_ZVBI="$(find ../src/zvbi/ -name '*.c')"
+@@ -10,7 +11,7 @@ SRC_HASH="$(find ../src/lib_hash/ -name '*.c')"
+ SRC_PROTOBUF="$(find ../src/protobuf-c/ -name '*.c')"
+ SRC_UTF8PROC="../src/utf8proc/utf8proc.c"
+ BLD_SOURCES="../src/ccextractor.c $SRC_CCX $SRC_GPAC $SRC_ZLIB $SRC_ZVBI $SRC_LIBPNG $SRC_HASH $SRC_PROTOBUF $SRC_UTF8PROC"
+-BLD_LINKER="-lm -zmuldefs -l tesseract -l lept"
++BLD_LINKER="-liconv -lnetwork -lm -zmuldefs -l tesseract -l lept"
+ 
+ ./pre-build.sh
+ 
+diff --git a/src/gpacmp4/os_file.c b/src/gpacmp4/os_file.c
+index 901e292..021c841 100644
+--- a/src/gpacmp4/os_file.c
++++ b/src/gpacmp4/os_file.c
+@@ -50,6 +50,14 @@
+ #include <errno.h>
+ #endif
+ 
++#ifdef __HAIKU__
++typedef off_t off64_t;
++
++#define ftello64(stream) ftello(stream)
++#define fseeko64(stream, offset, whence) fseeko(stream, offset, whence)
++#define fopen64(filename, type) fopen(filename, type)
++#endif
++
+ #endif
+ 
+ 
+diff --git a/src/lib_ccx/ocr.c b/src/lib_ccx/ocr.c
+index ebd7619..da190ad 100644
+--- a/src/lib_ccx/ocr.c
++++ b/src/lib_ccx/ocr.c
+@@ -142,8 +142,7 @@ void* init_ocr(int lang_index)
+ 	char* pars_vec = strdup("debug_file");
+ 	char* pars_values = strdup("/dev/null");
+ 
+-	ret = TessBaseAPIInit4(ctx->api, tessdata_path, lang, ccx_options.ocr_oem, NULL, 0, &pars_vec,
+-		&pars_values, 1, false);
++	ret = TessBaseAPIInit2(ctx->api, tessdata_path, lang, ccx_options.ocr_oem);
```

Comment:
This change isn't Haiku specific, why?

Reply:
Because Haiku's tesseract version does not have TessBaseAPIInit4.

Reply:
Indeed it appeared in version 3.04.00.